shim/src/intrinsics.js

```diff
@@ -0,0 +1,299 @@
+import Realm from './realm';
+import { getPrototypeOf } from './commons';
+
+/**
+ * Get the intrinsics from Table 7 & Annex B
+ * Named intrinsics: available as data properties of the global object.
+ * Anonymous intrinsics: not otherwise reachable by own property name traversal.
+ *
+ * https://tc39.github.io/ecma262/#table-7
+ * https://tc39.github.io/ecma262/#table-73
+ */
+export function getIntrinsics(sandbox) {
+  const { unsafeGlobal: g } = sandbox;
+
+  // Anonymous intrinsics.
+
+  const SymbolIterator = g.Symbol.iterator;
+
+  const ArrayIteratorInstance = new g.Array()[SymbolIterator]();
+  const ArrayIteratorPrototype = getPrototypeOf(ArrayIteratorInstance);
+  const IteratorPrototype = getPrototypeOf(ArrayIteratorPrototype);
+
+  const AsyncFunctionInstance = g.eval('(async function(){})');
+  const AsyncFunction = AsyncFunctionInstance.constructor;
+  const AsyncFunctionPrototype = AsyncFunction.prototype;
+
+  const GeneratorFunctionInstance = g.eval('(function*(){})');
+  const GeneratorFunction = GeneratorFunctionInstance.constructor;
+  const Generator = GeneratorFunction.prototype;
+  const GeneratorPrototype = Generator.prototype;
+
+  const hasAsyncIterator = typeof g.Symbol.asyncIterator !== 'undefined';
+
+  const AsyncGeneratorFunctionInstance = hasAsyncIterator && g.eval('(async function*(){})');
+  const AsyncGeneratorFunction = hasAsyncIterator && AsyncGeneratorFunctionInstance.constructor;
+  const AsyncGenerator = hasAsyncIterator && AsyncGeneratorFunction.prototype;
+  const AsyncGeneratorPrototype = hasAsyncIterator && AsyncGenerator.prototype;
+
+  const AsyncIteratorPrototype = hasAsyncIterator && getPrototypeOf(AsyncGeneratorPrototype);
+  const AsyncFromSyncIteratorPrototype = undefined; // Not reacheable.
+
+  const MapIteratorObject = new g.Map()[SymbolIterator]();
+  const MapIteratorPrototype = getPrototypeOf(MapIteratorObject);
+
+  const SetIteratorObject = new g.Set()[SymbolIterator]();
+  const SetIteratorPrototype = getPrototypeOf(SetIteratorObject);
+
+  const StringIteratorObject = new g.String()[SymbolIterator]();
+  const StringIteratorPrototype = getPrototypeOf(StringIteratorObject);
+
+  const ThrowTypeError = g.eval(
+    '(function () { "use strict"; return Object.getOwnPropertyDescriptor(arguments, "callee").get; })()'
+  );
+
+  const TypedArray = getPrototypeOf(g.Int8Array);
+  const TypedArrayPrototype = g.TypedArray.prototype;
+
+  // Named intrinsics
```

Comment:
The following object also includes the anonymous intrinsics from the previous section.